File: back_end/src/main/java/org/example/cmpe202_final/service/notification/NotificationService.java
package org.example.cmpe202_final.service.notification;

import lombok.AllArgsConstructor;
import org.example.cmpe202_final.model.notification.Notification;
import org.example.cmpe202_final.repository.notifications.NotificationRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@AllArgsConstructor
@Service
public class NotificationService {
    private final NotificationRepository repository;
    public List<Notification> getAllNotifications() {
        return repository.findAllByOrderByDateDesc();
    }

    public void insertNotification(Notification notification) {
        repository.save(notification);
    }
}

File: front_end/app/src/main/java/com/example/studentportal/notifications/service/NotificationService.kt
package com.example.studentportal.notifications.service
import com.example.studentportal.notifications.usecase.model.NotificationUseCaseModel
import retrofit2.Call
import retrofit2.http.GET

interface NotificationService {
    @GET("/notifications")
    fun fetchNotifications(): Call<List<NotificationUseCaseModel>>
}
